api/types/provisioning.go

```diff
@@ -817,3 +832,28 @@ func (a *ProvisionTokenSpecV2TPM) validate() error {
 	}
 	return nil
 }
+
+func (a *ProvisionTokenSpecV2Terraform) checkAndSetDefaults() error {
+	if len(a.Allow) == 0 {
+		return trace.BadParameter("the %q join method requires at least one token allow rule", JoinMethodTerraform)
+	}
+
+	if a.Audience == "" {
+		return trace.BadParameter("audience: should match the audience specified in TFC's TFC_WORKLOAD_IDENTITY_AUDIENCE variable")
+	}
+
+	for i, allowRule := range a.Allow {
+		orgSet := allowRule.OrganizationID != "" || allowRule.OrganizationName != ""
+		projectSet := allowRule.ProjectID != "" || allowRule.ProjectName != ""
```

Comment:
Are project names globally unique across TfCloud ? We should ensure we require at least one globally unique property here.

Reply:
Organization names are, at least. I think we could require that at least organization (name or ID) is set, then any combination of project/workspace?

Reply:
👍 I think I've done similar on another one of the join methods (always required org id/org name). Without it, it's super easy to make a join token that would allow a bad actor to create their own org, create a project w/ the same name, and access your cluster.